src/net/sourceforge/kolmafia/session/CrystalBallManager.java

```diff
@@ -135,12 +162,26 @@ private static String parseCrystalBallMonster(final String responseText) {
     return null;
   }
 
+  /*
+   * Quote from a /devster
+   *
+   * orb predictions are valid if:
+   * * your turnsplayed is <= the turnsplayed when you got the prediction + 1
+   * OR
+   * * your [lastadv] flag is the zone the prediction is in.
+   */
   public static void updateCrystalBallPredictions() {
-    if (KoLAdventure.lastVisitedLocation() == null) {
+    // We use the vanilla last adventure here as sometimes mafia does not recognize it
+    // And we still want to invalidate the predictions, even if mafia doesn't recognize it
+    String lastAdventureName = KoLAdventure.lastZoneName;
+
+    if (lastAdventureName == null) {
       return;
     }
 
-    String lastAdventureName = KoLAdventure.lastLocationName;
+    if (CrystalBallManager.predictions.isEmpty()) {
```

Comment:
minor: you could shift this to the very top of the function to avoid copying the `lastAdventureName` in the (hopefully common) case that there are no valid predictions because the user isn't using the crystal ball.

Reply:
Change made